tests/test_cdf_tk/test_auth.py

```diff
@@ -0,0 +1,203 @@
+    monkeypatch.setenv("CDF_CLUSTER", "bluefield")
+    monkeypatch.setenv("IDP_TOKEN_URL", "dummy")
+    monkeypatch.setenv("IDP_CLIENT_ID", "dummy")
+    monkeypatch.setenv("IDP_CLIENT_SECRET", "dummy")
+    monkeypatch.setenv("IDP_TENANT_ID", "dummy")
+
+    real_config = CDFToolConfig(cluster="bluefield", project="pytest-project")
+    # Build must always be executed from root of the project
+    cdf_tool = MagicMock(spec=CDFToolConfig)
+    cdf_tool.failed = False
+    cdf_tool.environment_variables.side_effect = real_config.environment_variables
+    # Load the rw-group and add it to the mock client as an existing resource
+    group = Group.load(yaml.safe_load((DATA_FOLDER / "rw-group.yaml").read_text()))
+    project_capabilities: ProjectCapabilityList = []
+    for cap in group.capabilities:
+        project_capabilities.append(ProjectCapability(capability=cap, project_scope=AllProjectsScope()))
+    inspect_result = TokenInspection(
+        subject="test@test.com",
+        projects=[ProjectSpec("pytest-project", groups=[1234567890])],
+        capabilities=ProjectCapabilityList(project_capabilities),
+    )
+
+    # Mock the get call to return the project info
+    def mock_get_json(*args, **kwargs):
+        mock = MagicMock()
+        mock.json.return_value = json.loads(Path(DATA_FOLDER / "project_info.json").read_text())
+        return mock
+
+    # Set the mock get call to return the project info
+    cognite_client_approval.client.get.side_effect = mock_get_json
+
+    # Set the side effect of the MagicMock to the real method
+    cognite_client_approval.client.iam.compare_capabilities.side_effect = IAMAPI.compare_capabilities
+
+    return CDFDataFixture(
+        cdf_tool=cdf_tool,
+        approval_client=cognite_client_approval,
+        data={
+            TokenInspection: inspect_result,
+            Group: GroupList([group]),
+            # NOTE! If you add more resources to be pre-loaded in the CDF project, add them here.
+        },
+    )
+
+
+def test_auth_verify_happypath(
+    file_regression,
+    cdf_data_fixture: CDFDataFixture,
+    capfd,
+):
+    # First add the pre-loaded data to the approval_client
+    for resource, data in cdf_data_fixture.data.items():
+        cdf_data_fixture.approval_client.append(resource, data)
+    # Then make sure that the CogniteClient used is the one mocked by
+    # the approval_client
+    cdf_data_fixture.cdf_tool.client = cdf_data_fixture.approval_client.mock_client
+    check_auth(cdf_data_fixture.cdf_tool, group_file=DATA_FOLDER / "rw-group.yaml")
+    out, _ = capfd.readouterr()
+    if platform.system() == "Windows":
+        # Windows console use different characters for tables in rich.
+        fullpath = SNAPSHOTS_DIR / f"{TEST_PREFIX}_auth_verify_happypath_windows.txt"
+    else:
+        fullpath = SNAPSHOTS_DIR / f"{TEST_PREFIX}_auth_verify_happypath.txt"
+    file_regression.check(out, encoding="utf-8", fullpath=fullpath)
```

Comment:
I would just make a helper for this
```suggestion
    file_regression.check(out, encoding="utf-8", fullpath=to_fullpath(f"{TEST_PREFIX}_auth_verify_happypath"))
```

Something like this
```
def to_fullpath(filename: str) -> Path:
    if platform.system() == "Windows":
        # Windows console use different characters for tables in rich.
        return SNAPSHOTS_DIR / f"{file_name}_windows.txt"
    else:
        return SNAPSHOTS_DIR / f"{file_name}.txt"